src/main/java/com/opentable/metrics/FloatingPointHistogram.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.metrics;
+
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Reservoir;
+import com.codahale.metrics.Snapshot;
+
+/**
+ * This class extends {@link Histogram} to include floating-point {@link #update} functions. Because the underlying
+ * {@link Reservoir}s store only {@code long}s, this class also takes a {@link #scale} by which floating-point
+ * values are <em>multiplied</em> before being rounded and stored as {@code long}s. The {@link Snapshot} returned
+ * wraps the underlying {@link Histogram#getSnapshot()} return value with an implementation that
+ * &ldquo;un-scales&rdquo; the returned values by <em>dividing</em> by the {@link #scale}.
+ *
+ * <p>
+ * This is essentially just the ol' &ldquo;convert to cents&rdquo; trick, but you may choose the magnitude of the
+ * {@link #scale} based on the expected range of floating-point values stored as well as the desired precision you
+ * wish to preserve.
+ *
+ * <p>
+ * You may still of course still call {@link #update(int)} or {@link #update(long)}; behavior in this case is
+ * undefined.
+ *
+ * <p>
+ * cf. <a href="https://github.com/dropwizard/metrics/issues/863">this issue</a>
+ */
+public class FloatingPointHistogram extends Histogram {
```

Comment:
Nifty.  Annoying that this is needed, but the implementation is elegant.  I wonder if this would make a good PR back to upstream dropwizard-metrics.

Reply:
Yeah, I wonder that too.